### PR TITLE
Remove `hash_strategy` from `Tree`.

### PR DESCRIPTION
**Pull request by pphaneuf**
Part of #2450.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: storage/cloudspanner/log_storage.go [hunk 1]
```diff
@@ -158,9 +158,6 @@ func (ls *logStorage) Snapshot(ctx context.Context) (storage.ReadOnlyLogTX, erro
 }
 
 func newLogCache(tree *trillian.Tree) (*cache.SubtreeCache, error) {
-	if s := tree.HashStrategy; s != trillian.HashStrategy_RFC6962_SHA256 {
-		return nil, fmt.Errorf("unknown hash strategy: %s", s)
-	}
 	return cache.NewLogSubtreeCache(defLogStrata, rfc6962.DefaultHasher), nil
 }
```
Comment (pavelkalinnikov) [inline storage/cloudspanner/log_storage.go:162]: Inline this?

File: storage/sql.go [hunk 2]
```diff
@@ -96,11 +96,6 @@ func ReadTree(row Row) (*trillian.Tree, error) {
 	} else {
 		return nil, fmt.Errorf("unknown TreeType: %v", treeType)
 	}
-	if hs, ok := trillian.HashStrategy_value[hashStrategy]; ok {
-		tree.HashStrategy = trillian.HashStrategy(hs)
-	} else {
-		return nil, fmt.Errorf("unknown HashStrategy: %v", hashStrategy)
-	}
```
Comment (pavelkalinnikov) [inline storage/sql.go:103]: After deleting this code, `hashStrategy` seems not used. Would here be a good place to verify that `hashStrategy == "RFC6962_SHA256"` for `[PREORDERED_]LOG` trees? As a safe-guard for the (unlikely) case that a log with a different hasher exists somewhere.

**Comment by pavelkalinnikov** [comment]
The "dumplib" test failure seems non-trivial. I wonder if it used a different hasher.

**Comment by pavelkalinnikov** [comment]
Or it looks like a formatting issue.

**Comment by pphaneuf** [comment]
Yeah, it seems like a formatting issue, which I find somewhat baffling?

Taking a look at your other comments now...

File: storage/sql.go (hunk 2 above)
Comment (pphaneuf) [reply storage/sql.go:103]: `hashStrategy` is still somewhat necessary for the SQL query (I didn't want to fiddle with those, seemed easier and safer to still read/write them the same way), but good call on having a check to see that it matches what we expect it to be.

I'm not sure I get why the "for `[PREORDERED_]LOG` trees" bit? The only "known" `HashStrategy` is `RFC6962_SHA256`, so just generally check for that?

Comment (pavelkalinnikov) [reply storage/sql.go]: Yeah, can make the check unconditional, since all trees are expected to be logs.

Comment (pphaneuf) [reply storage/sql.go]: Done, PTAL

**Comment by pphaneuf** [comment]
Increased overall coverage from 65.94% to 67.09%, but getting told off about not having enough incremental coverage? Deleting code is so thankless... 🤣

**Review by pavelkalinnikov — APPROVED** [review]

**Pull request merged**